src/pybind/mgr/diskprediction/module.py

```
@@ -0,0 +1,346 @@
+
+        for key, default in self.config_keys.items():
+            self.set_config_option(key, self.get_config(key, default))
+
+    def serve(self):
+        self.log.info('Starting diskprediction module')
+        self.run = True
+        self.status = DP_MGR_STAT_ENABLED
+
+        while self.run:
+            if self.get_configuration("diskprediction_config_mode").lower() in ['cloud', 'onpremise']:
+                enable_colud = True
+            if enable_colud and not self._activate_cloud:
+                self.start_cloud_disk_prediction()
+            elif not enable_colud and self._activate_cloud:
+                self.stop_cloud_disk_prediction()
+            self.event.wait(60)
+
+    def start_cloud_disk_prediction(self):
+        try:
+            if not self._activate_cloud:
+                for dp_task in DP_TASK:
+                    obj_task = dp_task(self)
+                    if obj_task:
+                        obj_task.run()
+                    else:
+                        raise Exception('failed to start task %s' % obj_task._task_name)
+                    self._tasks.append(obj_task)
+                self._activate_cloud = True
+                self.log.info('start cloud disk prediction')
+        except:
+            self.log.error('failed to start cloud disk prediction')
+            if self._tasks:
+                self._activate_cloud = True
+                self.stop_cloud_disk_prediction()
+
+    def stop_cloud_disk_prediction(self):
+        try:
+            if self._activate_cloud:
+                self.status = DP_MGR_STAT_DISABLED
+                while self._tasks:
+                    try:
+                        dp_task = self._tasks.pop()
+                        dp_task.terminate()
+                        del dp_task
+                    except:
+                        break
+                self._activate_cloud = False
+                self.log.info("stop cloud disk prediction")
+        except:
```

If there isn't any logic to handle an exception, don't catch it.  Just let it bubble up, there is error handling logic in ceph-mgr that notices unhandled exceptions and marks modules as errored when it happens.